api/server/gin_middlewares.go

```diff
@@ -100,11 +101,11 @@ func RegisterAPIViews(tagKeys []string, dist []float64) {
 
 	// default tags for request and response
 	reqTags := []tag.Key{pathKey, methodKey}
-	respTags := []tag.Key{pathKey, methodKey, statusKey, whodunitKey}
+	respTags := []tag.Key{pathKey, methodKey, statusKey, whodunitKey, fnFdkVersionKey}
```

Comment:
i forget the tangle here so probing a little, is there a way to hang the whodunitkey and the fdkversionkey only off of the api response for invoke? it's unnecessary for all other endpoints and could lead to confusion, adds excess data to metrics, etc - realize whodunitkey was my doing to begin w/, sorry

Reply:
The only way to not have the tags for specific apis, is to create a separate view for the invoke apis. I am worried about the downstream effect of this because this seems like a backwards incompatible change.

It results in metrics like this-
```
# HELP fn_api_invoke_latency Latency distribution of Invoke API requests
# TYPE fn_api_invoke_latency counter
fn_api_invoke_latency{blame="user",fn_fdk_version="fdk-go/0.0.1",method="POST",path="/invoke/:fn_id",status="200"} 6
fn_api_invoke_latency{blame="user",fn_fdk_version="fdk-python/0.1.11",method="POST",path="/invoke/:fn_id",status="200"} 6
# HELP fn_api_invoke_response_count Invoke API response count
# TYPE fn_api_invoke_response_count counter
fn_api_invoke_response_count{blame="user",fn_fdk_version="fdk-go/0.0.1",method="POST",path="/invoke/:fn_id",status="200"} 6
fn_api_invoke_response_count{blame="user",fn_fdk_version="fdk-python/0.1.11",method="POST",path="/invoke/:fn_id",status="200"} 6
# HELP fn_api_latency Latency distribution of API requests
# TYPE fn_api_latency histogram
fn_api_latency_bucket{method="GET",path="/v2/apps",status="200",le="1"} 0
fn_api_latency_bucket{method="GET",path="/v2/apps",status="200",le="10"} 11
fn_api_latency_bucket{method="GET",path="/v2/apps",status="200",le="50"} 12
fn_api_latency_bucket{method="GET",path="/v2/apps",status="200",le="100"} 12
fn_api_latency_bucket{method="GET",path="/v2/apps",status="200",le="250"} 12
fn_api_latency_bucket{method="GET",path="/v2/apps",status="200",le="500"} 12
fn_api_latency_bucket{method="GET",path="/v2/apps",status="200",le="1000"} 12
fn_api_latency_bucket{method="GET",path="/v2/apps",status="200",le="10000"} 12
fn_api_latency_bucket{method="GET",path="/v2/apps",status="200",le="60000"} 12
fn_api_latency_bucket{method="GET",path="/v2/apps",status="200",le="120000"} 12
fn_api_latency_bucket{method="GET",path="/v2/apps",status="200",le="+Inf"} 12
fn_api_latency_sum{method="GET",path="/v2/apps",status="200"} 37
fn_api_latency_count{method="GET",path="/v2/apps",status="200"} 12
fn_api_latency_bucket{method="GET",path="/v2/fns",status="200",le="1"} 0
fn_api_latency_bucket{method="GET",path="/v2/fns",status="200",le="10"} 12
fn_api_latency_bucket{method="GET",path="/v2/fns",status="200",le="50"} 12
fn_api_latency_bucket{method="GET",path="/v2/fns",status="200",le="100"} 12
fn_api_latency_bucket{method="GET",path="/v2/fns",status="200",le="250"} 12
fn_api_latency_bucket{method="GET",path="/v2/fns",status="200",le="500"} 12
fn_api_latency_bucket{method="GET",path="/v2/fns",status="200",le="1000"} 12
fn_api_latency_bucket{method="GET",path="/v2/fns",status="200",le="10000"} 12
fn_api_latency_bucket{method="GET",path="/v2/fns",status="200",le="60000"} 12
fn_api_latency_bucket{method="GET",path="/v2/fns",status="200",le="120000"} 12
fn_api_latency_bucket{method="GET",path="/v2/fns",status="200",le="+Inf"} 12
fn_api_latency_sum{method="GET",path="/v2/fns",status="200"} 31
fn_api_latency_count{method="GET",path="/v2/fns",status="200"} 12
# HELP fn_api_request_count Count of API requests started
# TYPE fn_api_request_count counter
fn_api_request_count{method="GET",path="/v2/apps"} 12
fn_api_request_count{method="GET",path="/v2/fns"} 12
fn_api_request_count{method="POST",path="/invoke/:fn_id"} 12
# HELP fn_api_response_count API response count
# TYPE fn_api_response_count counter
fn_api_response_count{method="GET",path="/v2/apps",status="200"} 12
fn_api_response_count{method="GET",path="/v2/fns",status="200"} 12
```

Reply:
@rdallman I compromised by retaining the invoke path in the original metrics but removing the fdk and blame tags, and creating new metrics for the fn invoke path so that the fdk and blame tags can be included.

Reply:
On second thought, I reverted the separate metrics for invoke vs all other apis. The compromise was confusing too. :| With the current limitation of opencensus only supporting a static list of tags,( though it's not ideal) having the tag be empty for the non-invoke apis is the best we can do.

Reply:
no worries - I think the limitation was 'inconsistent label cardinality' or something of the sort, I had forgotten how fun it all was. thanks